### ISSUE TITLE
zkvm: guaranteed optimization for `range` instruction with const expressions 

### ISSUE THREAD

**Issue by oleganza** [stellar/slingshot#310]
Current optimizations for constant Expressions and cleartext Constraints cover all arithmetic operations except for the `range:n` instruction. But this is the most costly of all of them, and has `n`-bits parameter to tune down that cost.

Suggestion:

1. Specify a guaranteed optimization for `range` instruction over _constant expressions_ such that the check is performed on the cleartext scalar bits: all bits in positions [64;255] must be zero.
2. Remove `:n` parameter, making the range work on full 64-bit integer range. This simplifies the design and provides better confidentiality for when instruction is used with actually secret data: amounts and time intervals.

**Comment by oleganza** [comment]
Guidelines:

1. `vm.rs` has `fn add_range_proof` [1](https://github.com/stellar/slingshot/blob/main/zkvm/src/vm.rs#L675-L689) that currently converts ZkVM `Expression` object to r1cs::LinearCombination and builds rangeproof constraints using `spacesuit` API. This should change to check if `Expression` is `Expression::Constant` [2](https://github.com/stellar/slingshot/blob/main/zkvm/src/constraints.rs#L27-L34) and then use the stored `ScalarWitness` to check its range. 
2. The `ScalarWitness` is implemented in [`zkvm/src/scalar_witness.rs`](https://github.com/stellar/slingshot/blob/main/zkvm/src/scalar_witness.rs#L14-L19) and contains either a `SignedInteger` (think of it as "u64 + sign", aka "i65", [defined in Spacesuit](https://github.com/stellar/slingshot/blob/main/spacesuit/src/signed_integer.rs#L9)) or full-size Scalar (of order `2^252+...`). The difference in underlying representation is so that the prover can have sanity check of their integer operations. But we should not leak this out to the Verifier.
3. So for the verifier, we need to convert the ScalarWitness into the Scalar (ignoring whether it was already a Scalar or was a compact SignedInteger), and check that it's in range with the same semantics as the Cloak's implementation, but in cleartext: checking that top 256-64=192 bits are all-zeroes (meaning that only bottom 64 bits are allowed to be whatever). I think the proper place for this bit checking is in a method on ScalarWitness: `fn ScalarWitness::in_range(&self)->bool`.

**Comment by oleganza** [comment]
TL;DR: verifier's view on Expression::Constant should be that it's "just a scalar", regardless if we have preserved the narrower SignedInteger inside or not. Therefore there **must not be any attempt** to switch on `ScalarWitness::Integer/::Scalar` to choose how to check the range. The ScalarWitness must be converted to plain Scalar and then the Scalar's bits must be checked in the similar manner as they are checked by ZK proof: bit by bit.
